hpa/README.md

```diff
@@ -0,0 +1,76 @@
+
+## Experiment lifecycle
+
+For each trial, we create a locust load test for the configuration tested using the following trial pod:
+
+```
+          template: # pod
+            spec:
+              containers:
+              - name: locust-trial-job
+                image: broadinstitute/python-requests
+                command: ["python", "/home/load_test.py"]
+                env:
+                - name: LOCUST_HOST
+                  valueFrom:
+                    configMapKeyRef:
+                      name: trialjob-env
+                      key: locust_host
+                - name: N_CLIENTS
+                  valueFrom:
+                    configMapKeyRef:
+                      name: trialjob-env
+                      key: n_clients
+                - name: HATCH_RATE
+                  valueFrom:
+                    configMapKeyRef:
+                      name: trialjob-env
+                      key: hatch_rate
+                - name: RUN_TIME
+                  valueFrom:
+                    configMapKeyRef:
+                      name: trialjob-env
+                      key: run_time
+                - name: MAX_LATENCY # max avg latency in ms
+                  valueFrom:
+                    configMapKeyRef:
+                      name: trialjob-env
+                      key: max_latency
+                - name: MAX_ERROR_RATE # max error rate in %
+                  valueFrom:
+                    configMapKeyRef:
+                      name: trialjob-env
+                      key: max_error_rate
+```
+
+The configuration for the load test is in `trialjob_configmap.yaml`. The load test is 180 seconds long, uses 100 clients that are added at a rate of 1 client/second. You can increase this rate to make sure that your HPA scales fast enough to increase in load.
+[However, with a kubernetes version lower than `v1.18` you cannot change the scaling policies through the `v2beta2` API](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/). In this experiment, we are going to optimize for the `target CPU utilization` that the HPA uses to scale the `voting-service`  deployment.
+
+`MAX_LATENCY` is the maximum latency in ms we want a trial to report. If locust measures an average latency over the length of the load test (aka trial) higher than this value, we will report this trial as failed. `MAX_ERROR_RATE` is the maximum error rate in % measured by locust for which we don't report the trial as failed. You can see the logic for this in `load_test.py`.
+
+Create a dedicated namespace and insert that namespace in `kustomization.yaml`, then run the experiment using.
```

Comment:
```suggestion
Create a dedicated namespace and insert that namespace in the `kustomization.yaml`, then run the experiment using:
```

Reply:
could we have them do 
`kubectl create namespace recipe` and then `kustomize build . | kubect apply -n recipe -f -` to simplify the steps/not have to clone and change the kustomize file?

we can still have the optionality of creating their own namespace.